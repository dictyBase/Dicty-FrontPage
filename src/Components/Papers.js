// @flow
import React from "react"
import { Flex } from "grid-styled"
import FontAwsome from "react-fontawesome"

import {
  PaperContainer,
  PaperTitle,
  Header,
  ListBox,
  ListItems,
  LeadText,
  MainContent,
  SourceContent,
  SourceTitle,
  Link,
  MoreLink,
} from "./styles"

/**
 * The Papers component that will be displayed on the right in every
 * react web application of [dictyBase](http://dictybase.org).
 */

const Papers = (props: {
  /** List of paper items */
  papers: Array<{
    "id": number,
    "author": string,
    "title": string,
    "journal": string,
  }>,
}) => {
  const text = props.papers.map((paper, index) =>
    <ListItems key={index}>
      <LeadText>
        {paper.author}
      </LeadText>
      <MainContent>
        <strong><em>{paper.title}</em></strong>
      </MainContent>
      <SourceContent>
        <SourceTitle>Journal: </SourceTitle>
          {paper.journal}
        <Link href={paper.link} target="new"> Pubmed</Link>
      </SourceContent>
    </ListItems>,
  )

  return (
    <PaperContainer>
      <Header>
        <Flex wrap>
            <PaperTitle>
                <FontAwsome name="paperclip fa-lg" />
            </PaperTitle>
            <PaperTitle>
                LATEST PAPERS
            </PaperTitle>
        </Flex>
      </Header>
      <ListBox>
        {text}
      </ListBox>
      <MoreLink>
        <FontAwsome name="plus" />
        <Link href="" alt="more papers">
          {" "}more papers{" "}
        </Link>
      </MoreLink>
    </PaperContainer>
  )
}

export default Papers
