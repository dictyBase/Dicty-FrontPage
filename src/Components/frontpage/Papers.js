// @flow
import React from "react"
import { Flex } from "rebass"
import FontAwesome from "react-fontawesome"

import {
  PaperContainer,
  PaperTitle,
  Header,
  ListBox,
  ListItems,
  LeadText,
  MainContent,
  SourceContent,
  SourceTitle,
  Link,
  MoreLink,
} from "../styles"

/**
 * The Papers component that will be displayed on the right in every
 * react web application of [dictyBase](http://dictybase.org).
 */

const Papers = (props: {
  /** List of paper items */
  papers: Array<{
    id: number,
    author: string,
    title: string,
    journal: string,
    link: string,
  }>,
}) => {
  const text = props.papers.map((paper, index) => (
    <ListItems key={index}>
      <LeadText>{paper.author}</LeadText>
      <MainContent>
        <strong>
          <em>{paper.title}</em>
        </strong>
      </MainContent>
      <SourceContent>
        <SourceTitle>Journal: </SourceTitle>
        {paper.journal}
        <Link href={paper.link} target="new">
          {" "}
          Pubmed
        </Link>
      </SourceContent>
    </ListItems>
  ))

  return (
    <PaperContainer>
      <Header>
        <Flex wrap>
          <PaperTitle>
            <FontAwesome name="paperclip fa-lg" />
          </PaperTitle>
          <PaperTitle>LATEST PAPERS</PaperTitle>
        </Flex>
      </Header>
      <ListBox>{text}</ListBox>
      <MoreLink>
        <FontAwesome name="plus" />
        <Link href="" alt="more papers">
          {" "}
          more papers{" "}
        </Link>
      </MoreLink>
    </PaperContainer>
  )
}

export default Papers
