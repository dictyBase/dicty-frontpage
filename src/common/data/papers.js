module.exports = [
  {
    author: "Tanaka, Jahan, Kondo, Nakano & Yumura (2019)",
    title:
      "Cytokinesis D is Mediated by Cortical Flow of Dividing Cells Instead of Chemotaxis.",
    journal: "Cells 8",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31108912",
  },
  {
    author:
      "Odell, Sikirzhytski, Tikhonenko, Cobani, Khodjakov & Koonce (2019)",
    title:
      "Force balances between interphase centrosomes, as revealed by laser ablation.",
    journal: "Mol. Biol. Cell mbcE19010034",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31067156",
  },
  {
    author:
      "Chen, Luck, Rabe, Dinh, Shaulsky, Nelson, Gershenzon, Dickschat, Kollner & Chen (2019)",
    title:
      "A terpene synthase-cytochrome P450 cluster in Dictyostelium discoideum produces a novel trisnorsesquiterpene.",
    journal: "Elife 8",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31063135",
  },
  {
    author: "Yamada & Schaap (2019)",
    title:
      "Cyclic AMP induction of Dictyostelium prespore gene expression requires autophagy.",
    journal: "Dev. Biol.",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31051160",
  },
  {
    author: "Saga, Iwade, Araki, Ishikawa & Kawata (2019)",
    title: "Analysis of DrkA kinase's role in STATa activation.",
    journal: "Genes Cells",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31002205",
  },
]
