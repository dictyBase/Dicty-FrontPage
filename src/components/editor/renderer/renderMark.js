// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
