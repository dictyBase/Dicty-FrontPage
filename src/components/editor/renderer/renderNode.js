// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
