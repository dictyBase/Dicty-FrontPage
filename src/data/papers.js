module.exports = [
  {
    author: "Xie X, Gutierrez A, Trofimov V, Szilágyi I, Soldati T, Bakker E.",
    title:
    "<strong><em>Charged Solvatochromic Dyes as Signal Transducers in pH Independent Fluorescent and Colorimetric Ion Selective Nanosensors.</em></strong>",
    journal: "Anal Chem. 2015 Sep 9.[Epub ahead of print] <a href='https://www.ncbi.nlm.nih.gov/pubmed/26352133' target='new'>Pubmed</a>",
  },
  {
    author: "Schmith A, Spaller T, Gaube F, Fransson Å, Boesler B, Ojha S, Nellen W, Hammann C, Söderbom F, Winckler T.",
    title:
    "<strong><em>A host factor supports retrotransposition of the TRE5-A population in Dictyostelium cells by suppressing an Argonaute protein.</em></strong>",
    journal: "Mob DNA. 2015;6:14. doi: 10.1186/s13100-015-0045-5. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26339297' target='new'>Pubmed</a> ",
  },
  {
    author: "Von Bülow J, Beitz E.",
    title:
    "<strong><em>Number and Regulation of Protozoan Aquaporins Reflect Environmental Complexity.</em></strong>",
    journal: "Biol Bull. 2015 Aug;229(1):38-46. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  {
    author: "Santarriaga S, Petersen A, Ndukwe K, Brandt A, Gerges N, Scaglione JB, Scaglione KM.",
    title:
    "<strong><em>The Social Amoeba Dictyostelium discoideum is Highly Resistant to Polyglutamine Aggregation.</em></strong>",
    journal: "J Biol Chem. 2015 Sep 1. doi:pii: jbc.M115.676247. [Epub ahead of print]<a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  {
    author: "Kim MK, Kim JH, Kim JS, Kang SO.",
    title:
    "<strong><em>Structure of the 34 kDa F-actin-bundling protein ABP34 from Dictyostelium discoideum.</em></strong>",
    journal: "Acta Crystallogr D Biol Crystallogr. 2015 Sep 1;71(Pt 9):1835-49. doi: 10.1107/S139900471501264X. Epub 2015 Aug 25. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  
]
