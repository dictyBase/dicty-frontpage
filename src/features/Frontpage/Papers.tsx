import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

const useStyles = makeStyles({
  container: {
    textAlign: "left",
    paddingLeft: "10px",
    paddingRight: "10px",
    paddingBottom: "10px",
    backgroundColor: "#eff8fb",
    borderRadius: "15px",
    marginBottom: "10px",
    maxHeight: "440px",
    overflow: "auto",
    "@media (max-width: 768px)": {
      height: "350px",
    },
  },
  title: {
    paddingLeft: "5px",
    color: "#086a87",
    fontSize: "20px",
    verticalAlign: "top",
    textAlign: "left",
  },
  header: {
    color: "black",
    fontSize: "20px",
    padding: "15px 30px 0px 35px",
    verticalAlign: "top",
    textAlign: "right",

    "@media (max-width: 767px)": {
      fontSize: "24px",
      textAlign: "right",
      padding: "20px 5px 20px 15px",
    },
  },
  listBox: {
    padding: "0px 25px 10px 25px",
    fontSize: "13px",
    marginBottom: "5px",
    marginTop: "12px",
    "@media (max-width: 992px) and (min-width: 767px)": {
      fontSize: "10px",
    },
    "@media (max-width: 768px)": {
      fontSize: "16px",
    },
  },
  listItem: {
    listStyle: "none",
    marginBottom: "10px",
  },
  leadText: {
    color: "#0b3861",
    paddingRight: "10px",
  },
  mainContent: {
    paddingRight: "10px",
  },
  sourceContent: {
    color: "#0b3861",
  },
  sourceTitle: {
    paddingTop: "7px",
    fontWeight: "bold",
    textAlign: "center",
  },
  link: {
    textDecoration: "none",
    color: "#428bca",
  },
  bottomLink: {
    color: "#0b3861",
    fontSize: "11px",
    fontStyle: "italic",
    fontWeight: "normal",
    textAlign: "center",
    paddingBottom: "10px",

    "@media (min-width: 1400px)": {
      paddingTop: "30px",
      fontSize: "12px",
    },
  },
})

type Paper = {
  author: string
  title: string
  journal: string
  link: string
}

type Props = {
  papers: Paper[]
}

/** Widget that displays the latest Dicty papers */

const Papers = ({ papers }: Props) => {
  const classes = useStyles()

  const text = papers.map((paper, index) => (
    <li className={classes.listItem} key={index}>
      <span className={classes.leadText}>{paper.author}</span>
      <span className={classes.mainContent}>
        <strong>
          <em>{paper.title}</em>
        </strong>
      </span>
      <br />
      <span className={classes.sourceContent}>
        <span className={classes.sourceTitle}>Journal: </span>
        {paper.journal}
        <a
          className={classes.link}
          href={paper.link}
          target="_blank"
          rel="noopener noreferrer">
          {" "}
          Pubmed
        </a>
      </span>
    </li>
  ))

  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <Grid container>
          <span className={classes.title}>
            <FontAwesomeIcon icon="paperclip" size="sm" />
          </span>
          <span className={classes.title}> LATEST PAPERS</span>
        </Grid>
      </div>
      <ul className={classes.listBox}>{text}</ul>
      <div className={classes.bottomLink}>
        {/* <FontAwesome name="plus" />
        <Link to="/papers" alt="more papers">
          {" "}
          more papers{" "}
        </Link> */}
      </div>
    </div>
  )
}

export default Papers
