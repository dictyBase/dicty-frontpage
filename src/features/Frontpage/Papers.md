```js
import papers from "data/papers"

;<Papers papers={papers} />
```
