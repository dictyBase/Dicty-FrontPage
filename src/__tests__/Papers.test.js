import React from "react"
import Papers from "../Components/Papers"
import renderer from "react-test-renderer"
import papers from "../././data/papers"

test("News snapshot test", () => {
  const component = renderer.create(<Papers papers={papers} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
