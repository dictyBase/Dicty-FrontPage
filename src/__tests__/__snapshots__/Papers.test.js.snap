// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Papers snapshot test 1`] = `
<div
  className="sc-gHboQg fFLpfM"
>
  <div
    className="sc-CtfFt ksgeeT"
  >
    <div
      className="sc-bdVaJa bDWFJH sc-htpNat jtWIOA sc-bdVaJa iHZvIS"
      is={null}
    >
      <span
        className="sc-gJWqzi jpqoRp"
      >
        <span
          aria-hidden={true}
          className="fa fa-paperclip fa-lg"
        />
      </span>
      <span
        className="sc-gJWqzi jpqoRp"
      >
        LATEST PAPERS
      </span>
    </div>
  </div>
  <ul
    className="sc-hGoxap bpTpFR"
  >
    <li
      className="sc-bHwgHz jiapwT"
    >
      <span
        className="sc-dTdPqK eONKOG"
      >
        Xie X, Gutierrez A, Trofimov V, Szilágyi I, Soldati T, Bakker E.
      </span>
      <span
        className="sc-eMigcr bKZfpz"
      >
        <strong>
          <em>
            Charged Solvatochromic Dyes as Signal Transducers in pH Independent Fluorescent and Colorimetric Ion Selective Nanosensors.
          </em>
        </strong>
      </span>
      <span
        className="sc-itybZL gfzDPh"
      >
        <span
          className="sc-fzsDOv dWbQIZ"
        >
          Journal: 
        </span>
        Anal Chem. 2015 Sep 9.[Epub ahead of print]
        <a
          className="sc-rBLzX jKkeFB"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26352133"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="sc-bHwgHz jiapwT"
    >
      <span
        className="sc-dTdPqK eONKOG"
      >
        Schmith A, Spaller T, Gaube F, Fransson Å, Boesler B, Ojha S, Nellen W, Hammann C, Söderbom F, Winckler T.
      </span>
      <span
        className="sc-eMigcr bKZfpz"
      >
        <strong>
          <em>
            A host factor supports retrotransposition of the TRE5-A population in Dictyostelium cells by suppressing an Argonaute protein.
          </em>
        </strong>
      </span>
      <span
        className="sc-itybZL gfzDPh"
      >
        <span
          className="sc-fzsDOv dWbQIZ"
        >
          Journal: 
        </span>
        Mob DNA. 2015;6:14. doi: 10.1186/s13100-015-0045-5.
        <a
          className="sc-rBLzX jKkeFB"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26339297"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="sc-bHwgHz jiapwT"
    >
      <span
        className="sc-dTdPqK eONKOG"
      >
        Von Bülow J, Beitz E.
      </span>
      <span
        className="sc-eMigcr bKZfpz"
      >
        <strong>
          <em>
            Number and Regulation of Protozoan Aquaporins Reflect Environmental Complexity.
          </em>
        </strong>
      </span>
      <span
        className="sc-itybZL gfzDPh"
      >
        <span
          className="sc-fzsDOv dWbQIZ"
        >
          Journal: 
        </span>
        Biol Bull. 2015 Aug;229(1):38-46.
        <a
          className="sc-rBLzX jKkeFB"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="sc-bHwgHz jiapwT"
    >
      <span
        className="sc-dTdPqK eONKOG"
      >
        Santarriaga S, Petersen A, Ndukwe K, Brandt A, Gerges N, Scaglione JB, Scaglione KM.
      </span>
      <span
        className="sc-eMigcr bKZfpz"
      >
        <strong>
          <em>
            The Social Amoeba Dictyostelium discoideum is Highly Resistant to Polyglutamine Aggregation.
          </em>
        </strong>
      </span>
      <span
        className="sc-itybZL gfzDPh"
      >
        <span
          className="sc-fzsDOv dWbQIZ"
        >
          Journal: 
        </span>
        J Biol Chem. 2015 Sep 1. doi:pii: jbc.M115.676247. [Epub ahead of print]
        <a
          className="sc-rBLzX jKkeFB"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="sc-bHwgHz jiapwT"
    >
      <span
        className="sc-dTdPqK eONKOG"
      >
        Kim MK, Kim JH, Kim JS, Kang SO.
      </span>
      <span
        className="sc-eMigcr bKZfpz"
      >
        <strong>
          <em>
            Structure of the 34 kDa F-actin-bundling protein ABP34 from Dictyostelium discoideum.
          </em>
        </strong>
      </span>
      <span
        className="sc-itybZL gfzDPh"
      >
        <span
          className="sc-fzsDOv dWbQIZ"
        >
          Journal: 
        </span>
        Acta Crystallogr D Biol Crystallogr. 2015 Sep 1;71(Pt 9):1835-49. doi: 10.1107/S139900471501264X. Epub 2015 Aug 25.
        <a
          className="sc-rBLzX jKkeFB"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
  </ul>
  <div
    className="sc-krDsej kbEGrA"
  >
    <span
      aria-hidden={true}
      className="fa fa-plus"
    />
    <a
      alt="more papers"
      className="sc-rBLzX jKkeFB"
      href=""
    >
       
      more papers
       
    </a>
  </div>
</div>
`;
